src/box/lua/feedback_daemon.lua

```diff
@@ -26,13 +26,105 @@ local function get_fiber_id(f)
     return fid
 end
 
-local function fill_in_feedback(feedback)
+local function detect_docker_environment()
+    local fh = fio.open('/proc/1/cgroup', {'O_RDONLY'})
+    if not fh then return false end
+
+    -- fh:read() doesn't read empty files
+    local big_enough_chunk = 4096
+    local ok = fh:read(big_enough_chunk)
+    fh:close()
+    if not ok then return false end
+
+    if not string.find(ok, 'docker') then return false end
+
+    return true
+end
+
+local function is_system_space(sp)
+    local sp_id = sp.id
+    if box.schema.SYSTEM_ID_MIN <= sp_id and sp_id <= box.schema.SYSTEM_ID_MAX then
+        return true
+    end
+
+    return false
+end
+
+local function fill_in_base_info(feedback)
     if box.info.status ~= "running" then
         return nil, "not running"
     end
     feedback.tarantool_version = box.info.version
     feedback.server_id         = box.info.uuid
     feedback.cluster_id        = box.info.cluster.uuid
+end
+
+local function fill_in_platform_info(feedback)
+    feedback.os        = jit.os
+    feedback.arch      = jit.arch
+    feedback.is_docker = detect_docker_environment()
+end
+
+local function fill_in_space_indices(feedback, sp)
+    if not sp.index[0] then return end
+
+    feedback.features.has_primary_index = true
+    local idx_count = 0
+    for _, idx in pairs(sp.index) do
+        for _, part in pairs(idx.parts) do
+            if part.path ~= nil then
+                feedback.features.has_jsonpath_index = true
+                break
+            end
+        end
+        if idx.func ~= nil then
+            feedback.features.has_functional_index = true
+        end
+        if idx.type == 'TREE' then
+            feedback.features.has_tree_index = true
+        elseif idx.type == 'HASH' then
+            feedback.features.has_hash_index = true
+        elseif idx.type == 'RTREE' then
+            feedback.features.has_rtree_index = true
+        elseif idx.type == 'BITSET' then
+            feedback.features.has_bitset_index = true
+        end
+        idx_count = idx_count + 1
+    end
+
+    if idx_count > 1 then
+        feedback.features.has_secondary_index = true
+    end
+end
+
+local function fill_in_features(feedback)
+    feedback.features = feedback.features or {}
+
+    local is_memtx, is_vinyl, is_temporary, is_local
+    for _, sp in pairs(box.space) do
+        local is_system = is_system_space(sp)
+        if not is_system then
+            if sp.engine == 'vinyl' then is_vinyl = true end
+            if sp.engine == 'memtx' then
+                if sp.temporary ~= nil then is_temporary = true end
+                is_memtx = true
+            end
+            if sp.is_local ~= nil then is_local = true end
+            fill_in_space_indices(feedback, sp)
+        end
+    end
+
+    feedback.features.has_temporary_spaces = is_temporary
+    feedback.features.has_local_spaces = is_local
+    feedback.features.memtx = is_memtx
+    feedback.features.vinyl = is_vinyl
```

Comment:
Should it be `has_memtx` and `has_vinyl` as well?

Reply:
I moved from flags to counters, so now it has different names. See updated PR description